src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+
+import java.time.LocalDate;
+
+/**
+ * Constructs a deadline task.
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
+
+    /**
+     * Constructs a Deadline with a description and due date.
+     *
+     * @param description The description of the deadline task.
+     * @param by          The due date of the deadline.
+     * @throws DukeException If the date string is not in a valid format.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        this.by = DateChecker.parseDate(by);
+    }
+
+
+    /**
+     * Returns the due date of the deadline.
+     * @return
+     */
+    public LocalDate getBy() {
+        return by;
+    }
+
+    /**
+     * Returns the description of the saved task.
+     *
+     * @return The description of the saved task.
+     */
+    @Override
+    public String saveData() {
```

Comment:
Since the return value is a String, the naming is slightly misleading. getData() or getFileData() would be more accurate